### PR TITLE
Potential fix for mmap fail with Invalid argument

### PR DESCRIPTION
**Pull request by HenryQuan**
This PR adds a potential fix for mmap error with invalid argument if `cryptid 0` indicates the encryption is gone.

The reason of the invalid arguments comes from the offset based on the documentation:
```
The contents of a file mapping (as opposed to an anonymous
    mapping; see MAP_ANONYMOUS below), are initialized using length
    bytes starting at offset offset in the file (or other object)
    referred to by the file descriptor fd.  offset must be a multiple
    of the page size as returned by sysconf(_SC_PAGE_SIZE).
```

The offset is now aligned with the page size. Some apps may work because the offset is indeed the page size by chance. It fails for many because it can be larger than the page size, but not a multiple of it.

Additionally, the script for iOS now supports sending to the device directly if the local IP address if provided for faster debugging and testing.

I have only done some testing on iPad OS 16.7. So far, the following apps are working:
- YouTube
- Blue Archive (JP & CN)
- TradingView
- Brave Browser
- Tiny Wings

In summary, this PR potentially addresses the following issues:
- Invalid arguments with mmap
   - Align the offset based on the page size
- Operation not permitted with mmap
   - Load the binary into the memory as suggested in https://github.com/NyaMisty/fouldecrypt/issues/15 by @dlevi309 
   - Alternatively, launch the app in the foreground while dumping it
- Adjust the alignment to if required (memcpy)
- Cleanup the output folder if it exists

It is now working with the Rootless jailbreak by palera1n.

### CODE REVIEW

**Comment by HenryQuan** [comment]
While this works for the main exetuable, it doesn't work for frameworks. For frameworks, the permission flag should be PROT_READ without the PROT_EXEC. However, I am not sure how to detect framework vs the main executable.

**Comment by HenryQuan** [comment]
I have added another potential fix for Operation not permitted. However, some games/apps will stuck in memcpy after mremap_encrypted was succesful. I am investigating this issue.

**Comment by HenryQuan** [comment]
The memcpy issue is now ressolved by updating the alignment of base. However, it is more like a hack because we may not  really know the real alignment in the memory.

**Comment by paradiseduo** [comment]
good job，thanks for your work ~~
